cmd/xeniad/handlers/pattern.go

```diff
@@ -0,0 +1,83 @@
+// Package handlers contains the handler logic for processing requests.
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/ardanlabs/kit/db"
+	"github.com/ardanlabs/kit/web/app"
+	"github.com/coralproject/shelf/internal/wire/pattern"
+)
+
+// patternHandle maintains the set of handlers for the pattern api.
+type patternHandle struct{}
+
+// Pattern fronts the access to the pattern service functionality.
+var Pattern patternHandle
+
+//==============================================================================
+
+// List returns all the existing patterns in the system.
+// 200 Success, 404 Not Found, 500 Internal
+func (patternHandle) List(c *app.Context) error {
+	ps, err := pattern.GetAll(c.SessionID, c.Ctx["DB"].(*db.DB))
+	if err != nil {
+		if err == pattern.ErrNotFound {
+			err = app.ErrNotFound
+		}
+		return err
+	}
+
+	c.Respond(ps, http.StatusOK)
+	return nil
+}
+
+// Retrieve returns the specified Pattern from the system.
+// 200 Success, 400 Bad Request, 404 Not Found, 500 Internal
+func (patternHandle) Retrieve(c *app.Context) error {
+	p, err := pattern.GetByType(c.SessionID, c.Ctx["DB"].(*db.DB), c.Params["type"])
+	if err != nil {
+		if err == pattern.ErrNotFound {
+			err = app.ErrNotFound
+		}
+		return err
+	}
```

Comment:
Not sure of this pattern. Would almost want to move that to an error handling middleware which would add this if this is a reoccurring pattern.